src/KNNClassifier/index.js

```diff
@@ -162,29 +164,37 @@ class KNN {
     await io.saveBlob(JSON.stringify({ dataset, tensors }), fileName, 'application/octet-stream');
   }
 
-  load(path, callback) {
-    io.loadFile(path, (err, data) => {
-      if (data) {
-        const { dataset, tensors } = data;
-        this.mapStringToIndex = Object.keys(dataset).map(key => dataset[key].label);
-        const tensorsData = tensors
-          .map((tensor, i) => {
-            if (tensor) {
-              const values = Object.keys(tensor).map(v => tensor[v]);
-              return tf.tensor(values, dataset[i].shape, dataset[i].dtype);
-            }
-            return null;
-          })
-          .reduce((acc, cur, j) => {
-            acc[j] = cur;
-            return acc;
-          }, {});
-        this.knnClassifier.setClassifierDataset(tensorsData);
-        if (callback) {
-          callback();
-        }
+  load(pathOrData, callback) {
+    if (typeof pathOrData === 'object') {
+      this[loadData](pathOrData, callback);
+    } else {
+      io.loadFile(pathOrData, (err, data) => {
+        this[loadData](data, callback);
+      });
+    }
+  }
+
+  [loadData](data, callback) {
```

Comment:
This would be more standard or is there something I'm missing here?

```suggestion
  loadData(data, callback) {
```